src/templater.rs

```diff
@@ -378,6 +378,41 @@ impl<C, T: Template<C>> TemplateProperty<C> for PlainTextFormattedProperty<T> {
     }
 }
 
+/// Renders a list of template properties with the given separator.
+pub struct FormattablePropertyListTemplate<P, S> {
+    property: P,
+    separator: S,
+}
+
+impl<P, S> FormattablePropertyListTemplate<P, S> {
+    pub fn new<C>(property: P, separator: S) -> Self
+    where
+        P: TemplateProperty<C>,
```

Comment:
It might be nice to have a comment here that expands on what you wrote in the commit description. Something like

> In theory, `P` serves the role of `Template<Vec<C>>`, but it is difficult to implement `Template<Vec<C>>` when `C` may not have `Clone`. So, `P` is actually a `TemplateProperty<C, Output: IntoIterator<Template<()>>`. 

(I'm not sure if this explanation is exactly correct as I wrote it).

Reply:
To state the obvious, I wish there was a way to make more of these types have `Clone`, but I don't have a better solution and I'm sure you've thought about that plenty.

Reply:
The tricky part is the separator is not of `Template<()>` type.

Added doc comment:
```
Each template property can be extracted as a context-less value, but the
separator takes a context of type `C`.
```